### DISCUSSION TITLE
How to perform a MIDI program change

### DISCUSSION THREAD

**Discussion by garthholmes** [CoderLine/alphaTab#983] [Q&A]
Hi,

I have program changes in the .gp file, but they all play as piano and I couldn't find a way to change program in the Docs.

Hope someone can help.

Regards

**Comment by Danielku15** [comment]
Can you provide your test file? Generally Instrument Changes should be [supported](https://github.com/CoderLine/alphaTab/blob/cbecf6f1ecd4e14ad3ea17b557cb9fed01d711ce/src/importer/GpifParser.ts#L345) in GP files but there might be something special to your file we either not support or Arobas changed in a certain version.

**Comment by garthholmes** [comment]
Hi Daniel,

Thank you for responding.

I have yet to try it on a later version of AlphaTab however I will check the version of the software in question prior.

Thanking you
